components/Map.js

```diff
@@ -1,49 +1,67 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import ExternalLink from './ExternalLink';
 import { FormattedMessage } from 'react-intl';
 
-class Map extends React.Component {
-  static propTypes = {
-    lat: PropTypes.number,
-    long: PropTypes.number,
-    address: PropTypes.string,
-  };
-
-  onMapCreated(map) {
-    map.setOptions({
-      disableDefaultUI: false,
-    });
-  }
-
-  makeBbox(long, lat, zoomValue) {
-    return [long - zoomValue, lat - zoomValue, long + zoomValue, lat + zoomValue];
-  }
-
-  render() {
-    const { lat, long } = this.props;
-    const bbox = this.makeBbox(long, lat, 0.003);
-
-    return (
-      <div style={{ width: '100%', height: '100%' }}>
-        <iframe
-          width={'100%'}
-          height={'100%'}
-          frameBorder="0"
-          scrolling="no"
-          src={`https://www.openstreetmap.org/export/embed.html?bbox=${bbox}&marker=${lat}%2C${long}&layers=ND`}
-        ></iframe>
-        <br />
-
-        <ExternalLink
-          openInNewTab
-          href={`https://www.openstreetmap.org/?mlat=${lat}&amp;mlon=${long}#map=16/${lat}/${long}`}
-        >
-          <FormattedMessage id="map.viewLarger" defaultMessage="View Larger Map" />
-        </ExternalLink>
-      </div>
-    );
-  }
-}
+const tile2Long = (tile, zoom) => {
+  return (tile / Math.pow(2, zoom)) * 360 - 180;
+};
+
+const tile2Lat = (tile, zoom) => {
+  const n = Math.PI - (2 * Math.PI * tile) / Math.pow(2, zoom);
+  return (180 / Math.PI) * Math.atan(0.5 * (Math.exp(n) - Math.exp(-n)));
+};
+
+const long2tile = (long, zoom) => {
```

Comment:
These formulas are not easy to read, let's add a link to the docs to help the next developer that will read this

```suggestion
// See https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames#ECMAScript_(JavaScript/ActionScript,_etc.)
const long2tile = (long, zoom) => {
```

Reply:
Prefect, I would add this.
Took me a while to wrap my head around this also.

Reply:
Yes, basically what fixed the issue if to set the iframe url after component has mounted.
When page is ready iframe can call `OpenStreetMap` and generate the `map tiles`.